src/Riok.Mapperly/Emit/Syntax/SyntaxFactoryHelper.SymbolDeclaration.cs

```diff
@@ -28,4 +28,27 @@ public ClassDeclarationSyntax Class(string name, SyntaxTokenList modifiers, Synt
             .WithAttributeLists(isPartial ? new SyntaxList<AttributeListSyntax>() : GeneratedCodeAttributeList())
             .AddLeadingLineFeed(Indentation);
     }
+
+    public TypeDeclarationSyntax TypeDeclaration(TypeDeclarationSyntax syntax, SyntaxList<MemberDeclarationSyntax> members)
+    {
+        var name = Identifier(syntax.Identifier.ValueText);
+        TypeDeclarationSyntax type = syntax switch
+        {
+            ClassDeclarationSyntax => ClassDeclaration(name),
+            StructDeclarationSyntax => StructDeclaration(name),
+            InterfaceDeclarationSyntax => InterfaceDeclaration(name),
+            RecordDeclarationSyntax => RecordDeclaration(Token(SyntaxKind.RecordKeyword), name),
+            _ => throw new NotSupportedException("Unsupported type declaration syntax.")
```

Comment:
To simplify troubleshooting it may be helpful to include the type of the `syntax` in the error message.

Reply:
Makes sense.